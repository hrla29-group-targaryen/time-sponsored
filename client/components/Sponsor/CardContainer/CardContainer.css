.sponsor-card-container{
	display: flex;
	align-items: center;
}
.sponsor-card-container-sub{
	display: flex;
	justify-content: left;
}

.sponsor-card-left{
	position: relative;
	left: 25px;
}

.sponsor-card-left button{
	color: #0070eb;
	background-color: white;
    border-radius: 50%;
    border: none;
    font-size: 32px;
    cursor: pointer;
    box-shadow: 2px 2px rgba(0,0,0,0.2);
    width:35px;
    height:35px;
}

.sponsor-card-right{
	position: relative;
	left: -25px;
}

.sponsor-card-right button{
	color: #0070eb;
	background-color: white;
    border-radius: 50%;
    border: none;
    font-size: 32px;
    cursor: pointer;
    box-shadow: 2px 2px rgba(0,0,0,0.2);
    width:35px;
    height:35px;
}

.sponsor-btn-invisible{
	visibility: hidden;
}