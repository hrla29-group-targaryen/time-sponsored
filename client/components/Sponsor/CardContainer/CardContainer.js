import React from 'react';
import Card from './Card/Card.js';
import './CardContainer.css';
import axios from 'axios';

class CardContainer extends React.Component{
	constructor(props){
		super(props)
		this.state={
			group1: [],
			group2: [],
			group3: [],
			group4: [],
			current: 'group1'
		}
		this.getSponsoredData=this.getSponsoredData.bind(this)
		this.splitData=this.splitData.bind(this)
		this.changeDataSet=this.changeDataSet.bind(this)
	}

	componentDidMount(){
		this.getSponsoredData()
	}

	getSponsoredData(){
		const id = window.location.href.split("id=")[1]
	  	axios.get(`http://localhost:3000/api/sponsored/${id}`)
	  	.then(({data}) => this.splitData(data))
	  	.catch(e=>console.log(e))
	}

	splitData(data){
		if (data !== null) {
			let group1 = data.slice(0,3)
			let group2 = data.slice(3,6)
			let group3 = data.slice(6,9)
			let group4 = data.slice(9,12)
			this.setState({group1, group2, group3, group4})
		}
	}

	changeDataSet(input){
		let current = this.state.current
		let counter = Number(current[5])
		if (input === 'prev') {
			if(current !== 'group1') {
				counter -= 1
				let newCurr = `group${counter}`
				this.setState({current:newCurr})
			}
		} else if (input === 'next') {
			if(current !== 'group4') {
				counter += 1
				let newCurr = `group${counter}`
				this.setState({current:newCurr})
			}
		}
	}

	render(){
		let slicedData = (this.state.current === 'group1') 
		? this.state.group1 
		: (this.state.current === 'group2')
		? this.state.group2
		: (this.state.current === 'group3')
		? this.state.group3
		: this.state.group4

		let leftButtonStyle = (this.state.current === 'group1') ? "sponsor-card-left sponsor-btn-invisible" : "sponsor-card-left"
		let rightButtonStyle = (this.state.current === 'group4') ? "sponsor-card-right sponsor-btn-invisible" : "sponsor-card-right"

		return(
			<section className="sponsor-card-container">

				<div className={leftButtonStyle}>
					<button onClick={()=>this.changeDataSet("prev")}><i className="fas fa-angle-left"></i></button>
				</div>

				<div className="sponsor-card-container-sub">
					{
						(slicedData) &&
						slicedData.map((item,i)=> <Card key={i} item={item} />)
					}
				</div>

				<div className={rightButtonStyle}>
					<button onClick={()=>this.changeDataSet("next")}><i className="fas fa-angle-right"></i></button>
				</div>

			</section>
			)
	}
}

export default CardContainer;