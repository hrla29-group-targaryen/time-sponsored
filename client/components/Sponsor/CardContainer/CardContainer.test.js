import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import CardContainer from './CardContainer';

describe('testing CardContainer component', ()=> {

	it('should grab a snapshot of the component.', ()=> {
		const component = renderer.create(<CardContainer />)
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	})
	
	it('should return a single-node wrapper.', ()=> {
		expect(shallow(<CardContainer />).length).toEqual(1)
	})

	it('should check state', ()=> {
		let wrapper = shallow(<CardContainer />)
		expect(wrapper.state().group1).toEqual([])
		expect(wrapper.state().group2).toEqual([])
		expect(wrapper.state().group3).toEqual([])
		expect(wrapper.state().group4).toEqual([])
		expect(wrapper.state().current).toEqual('group1')	
	})

	//To check if componentDidMount() calls the 'getSponsoredData' function
	it('should render componentDidMount()', ()=> {
		let wrapper = shallow(<CardContainer />)
		let instance = wrapper.instance()
		jest.spyOn(instance, 'getSponsoredData')
		instance.componentDidMount()
		expect(instance.getSponsoredData).toHaveBeenCalledTimes(1)
	})

	//To check if componentDidMount() calls the 'getSponsoredData' function
	it('should check splitData()', ()=> {
		let wrapper = shallow(<CardContainer />)
		let fake_data = [{id:1},{id:2},{id:3},{id:4},{id:5},{id:6},{id:7},{id:8},{id:9},{id:10},{id:11},{id:12}]
		wrapper.instance().splitData(fake_data)
		expect(wrapper.state().group1).toEqual([{id:1},{id:2},{id:3}])
		expect(wrapper.state().group2).toEqual([{id:4},{id:5},{id:6}])
		expect(wrapper.state().group3).toEqual([{id:7},{id:8},{id:9}])
		expect(wrapper.state().group4).toEqual([{id:10},{id:11},{id:12}])
	})

	it('should check sponsor-card-container-button-prev button', ()=> {
		let wrapper = shallow(<CardContainer />)
		let instance = wrapper.instance()
		jest.spyOn(instance, 'changeDataSet')
		wrapper.find('#sponsor-card-container-button-prev').simulate('click')
		expect(instance.changeDataSet).toHaveBeenCalledTimes(1)
		wrapper.find('#sponsor-card-container-button-next').simulate('click')
		expect(instance.changeDataSet).toHaveBeenCalledTimes(2)
	})

	it('should check changeDataSet', ()=> {
		let wrapper = shallow(<CardContainer />)	
		expect(wrapper.instance().changeDataSet('prev')).toEqual(undefined)
		expect(wrapper.instance().changeDataSet('next')).toEqual(undefined)
	})
})
