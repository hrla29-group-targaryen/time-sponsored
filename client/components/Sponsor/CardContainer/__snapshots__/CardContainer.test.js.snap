// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing CardContainer component should grab a snapshot of the component. 1`] = `
<section
  className="sponsor-card-container"
>
  <div
    className="sponsor-card-left sponsor-btn-invisible"
  >
    <button
      id="sponsor-card-container-button-prev"
      onClick={[Function]}
    >
      <i
        className="fas fa-angle-left"
      />
    </button>
  </div>
  <div
    className="sponsor-card-container-sub"
  />
  <div
    className="sponsor-card-right"
  >
    <button
      id="sponsor-card-container-button-next"
      onClick={[Function]}
    >
      <i
        className="fas fa-angle-right"
      />
    </button>
  </div>
</section>
`;
